Unable to migrate contracts prior to compiling Typescript

When following the steps in the README, after running the Ganache testnet, if I then run `yarn truffle:migrate`, it outputs the following errors, because the README doesn't tell me that I need to explicitely compile the Typescript code.

```
yarn run v1.22.4
$ cd ./lib/mStable-subgraph && yarn truffle:migrate
$ cd lib/mStable-contracts && yarn migrate
$ truffle migrate --network development --reset

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.



Migrations dry-run (simulation)
===============================
> Network name:    'development-fork'
> Network id:      3
> Block gas limit: 7984452 (0x79d544)


1_main.js
=========

/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/dist/index.js:175
            throw error;
            ^

TSError: ⨯ Unable to compile TypeScript:
migrations/src/1_initial_migration.ts:9:20 - error TS2503: Cannot find namespace 'Truffle'.

9         artifacts: Truffle.Artifacts;
                     ~~~~~~~
migrations/src/1_initial_migration.ts:2:22 - error TS6053: File '/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/types/generated/index.d.ts' not found.

2 /// <reference path="../../types/generated/index.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
migrations/src/1_initial_migration.ts:3:22 - error TS6053: File '/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/types/generated/types.d.ts' not found.

3 /// <reference path="../../types/generated/types.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    at createTSError (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:245:12)
    at reportTSError (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:249:19)
    at getOutput (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:357:34)
    at Object.compile (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:415:32)
    at Module.m._compile (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:493:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/ts-node/src/index.ts:496:12)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at require (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/require/require.js:54:1)
    at /Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/migrations/1_main.js:4:26
    at Script.runInContext (vm.js:130:18)
    at Script.runInNewContext (vm.js:135:17)
    at Object.file (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/require/require.js:92:1)
    at Migration._load (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/migration.js:43:1)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at Migration.run (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/migration.js:167:1)
    at Object.runMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:148:1)
    at Object.runFrom (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:110:1)
    at Object.runAll (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:114:1)
    at Object.run (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:79:1)
    at runMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:253:1)
    at setupDryRunEnvironmentThenRunMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:241:1)
    at /Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:207:1
Truffle v5.1.22 (core: 5.1.22)
Node v12.18.3
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

But if I transpile the Typescript with:
```
cd ./lib/mStable-subgraph/lib/mStable-contracts && \
  yarn run generate-typings && \
  cd ../../../../
```

Then run `yarn truffle:migrate` again, it outputs:

```
yarn run v1.22.4
$ cd ./lib/mStable-subgraph && yarn truffle:migrate
$ cd lib/mStable-contracts && yarn migrate
$ truffle migrate --network development --reset

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.



Migrations dry-run (simulation)
===============================
> Network name:    'development-fork'
> Network id:      3
> Block gas limit: 7984452 (0x79d544)


1_main.js
=========

   Deploying 'Migrations'
   ----------------------
==============================================
Generating mock bAssets and lending platforms
==============================================


Error:  *** Deployment Failed ***

"Migrations" exceeded the block limit (with a gas value you set).
   * Block limit:  7984452 (0x79d544)
   * Gas sent:     8000000 (0x7a1200)
   * Try:
      + Sending less gas.
      + Setting a higher network block limit if you are on a
        private network or test client (like ganache).

    at /Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/deployer/src/deployment.js:364:1
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at Migration._deploy (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/migration.js:70:1)
    at Migration._load (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/migration.js:57:1)
    at Migration.run (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/migration.js:167:1)
    at Object.runMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:148:1)
    at Object.runFrom (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:110:1)
    at Object.runAll (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:114:1)
    at Object.run (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/migrate/index.js:79:1)
    at runMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:253:1)
    at setupDryRunEnvironmentThenRunMigrations (/Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:241:1)
    at /Users/ls/code/src/ltfschoen/mStable-app/lib/mStable-subgraph/lib/mStable-contracts/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:207:1
Truffle v5.1.22 (core: 5.1.22)
Node v12.18.3
```

Because I ran Ganache with the following:
```
ganache-cli \
  --account="0x209c205f333b5a65cc428589a51bd9f2621e2fc01de1b02dbf8c0f0b68e4974e, 50471238800000000000" \
  --account="0x0edb559026c8f779be17b4c9d8e4dfc14bead6592241de4d6612f77769327f7f, 100471238800000000000" \
  --unlock "0x209c205f333b5a65cc428589a51bd9f2621e2fc01de1b02dbf8c0f0b68e4974e" \
  --unlock "0x0edb559026c8f779be17b4c9d8e4dfc14bead6592241de4d6612f77769327f7f" \
  --port 7545 \
  --hostname localhost \
  --seed '0x209c205f333b5a65cc428589a51bd9f2621e2fc01de1b02dbf8c0f0b68e4974e' \
  --debug true \
  --mem true \
  --mnemonic 'end sleep vote expire arctic magic crack wrap toddler lizard acoustic owner' \
  --db './db/chain_database' \
  --verbose \
  --networkId=3 \
  --gasLimit=7984452 \
  --gasPrice=20000000000;
```

So I just changed `  --gasLimit=7984452 \` to `  --gasLimit=8000000 \` and it migrated successfully.

--- a/scripts/provision.sh
+++ b/scripts/provision.sh
@@ -6,10 +6,4 @@ git submodule update --init
 echo "Installing contracts"
 
 cd ./lib/mStable-subgraph
-git submodule update --init
-
-cd ./lib/mStable-contracts
-yarn install
-
-echo "Compiling contracts"
-yarn compile
+yarn run provision

